src/modules/sdlog2/sdlog2.c

```diff
@@ -1822,6 +1847,31 @@ int file_copy(const char *file_old, const char *file_new)
 	return OK;
 }
 
+int check_free_space()
+{
+	/* use statfs to determine the number of blocks left */
+	FAR struct statfs statfs_buf;
+	if (statfs(mountpoint, &statfs_buf) != OK) {
+		warnx("could not determine statfs");
+		return ERROR;
+	}
+
+	/* use a threshold of 4 MiB */
+	if (statfs_buf.f_bavail < (int)(4*1024*1024/statfs_buf.f_bsize)) {
+		warnx("no more space on MicroSD (less than 4 MiB)");
```

Comment:
This and many other locations would benefit from a all_log_critical() or similarly named macro / call which prints to the console and logs to mavlink. We're tight on space and duplicated strings cost kilobytes.
Let me push a prototype for that in a branch.